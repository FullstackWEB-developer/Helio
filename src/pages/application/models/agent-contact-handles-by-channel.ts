import {ChannelTypes} from '@shared/models';

export interface AgentContactHandlesByChannel {
    agent: string;
    channel: ChannelTypes;
    startInterval?: Date;
    endInterval?: Date;
    contactsHandled?: number
}
