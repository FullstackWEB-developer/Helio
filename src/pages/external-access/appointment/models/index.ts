export * from './appointment-type.model';
export * from './appointment-slot.model';
export * from './appointment.model';
export * from './appointment-cancellation.model';
export * from './appointment-cancel-reason.model';
export * from './appointment-slot-request.model';
