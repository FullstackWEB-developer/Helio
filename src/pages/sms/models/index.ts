export * from "./sms-new-message-steps";
export * from "./sms-notification-data";
export * from "./sms-query-type";
