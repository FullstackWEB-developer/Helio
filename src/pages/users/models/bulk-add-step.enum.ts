export enum BulkAddStep {
    Selection = 1,
    RolePicking,
    ProviderMapping,
    Review
}