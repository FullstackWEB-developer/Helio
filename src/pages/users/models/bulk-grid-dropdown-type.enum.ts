export enum BulkGridDropdownType {
    Role,
    ProviderMapping
}