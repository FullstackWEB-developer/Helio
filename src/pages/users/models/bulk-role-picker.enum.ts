export enum BulkRolePicker {
    Group,
    Individual
}