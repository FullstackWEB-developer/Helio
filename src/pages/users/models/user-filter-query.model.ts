export interface UserQueryFilter {
    searchText?: string;
    jobTitle?: string;
    departments?: string;
    roles?: string;
    invitationStatuses?: string;
    statuses?: string;
    rolesUnassigned?: string;
}