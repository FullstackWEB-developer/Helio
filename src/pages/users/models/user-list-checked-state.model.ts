import {CheckboxCheckEvent} from '@components/checkbox/checkbox';
import {UserDetailStatus, UserInvitationStatus} from '@shared/models';

export interface UserListCheckedState {
    checkboxCheckEvent: CheckboxCheckEvent;
    userStatus: UserDetailStatus;
    userInvitationStatus?: UserInvitationStatus;
    userEmail: string;
}