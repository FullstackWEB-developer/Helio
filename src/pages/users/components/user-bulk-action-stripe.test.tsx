import i18n from '../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {fireEvent, render, wait, waitFor} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import UserBulkActionStripe from './user-bulk-action-stripe';
import { act } from 'react-dom/test-utils';
import Router from "react-router";
import Api from '@shared/services/api';
import { ExternalUser, InviteUserModel, PagedList, SelectExternalUser, UserActiveDirectory } from '@shared/models';
import { BulkAddStep } from '../models/bulk-add-step.enum';
jest.setTimeout(30000);
jest.mock('react-router', () => ({
    ...jest.requireActual('react-router'),
    useParams: jest.fn().mockReturnValue({ type: "ticket-department" }),
  }))
describe("UserBulkActionStripe tests", () => {

    let container: HTMLDivElement | null;
    let mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        usersState: {
            selectedExternalUsers: [
                {
                    inviteUserModel: {
                        email: "string0",
                        firstName: "string0",
                        lastName: "string0",
                        name: "string0",
                        providerId: "string0",
                        roles: ["string0"]
                    } as InviteUserModel,
                    id: "string0",
                    info: {
                        azureId: "string0",
                        department: "Test",
                        jobTitle: "Test"
                    } as ExternalUser
                }
            ] as SelectExternalUser[],
            bulkPaging: {
                page: 1,
                pageSize: 25,
                totalCount: 2,
                totalPages: 1
            },
            selectedUsersLocalPagination: {
                page: 1,
                pageSize: 1,
                totalCount: 2,
                totalPages: 2
            },
            externalDepartments: ["Test", "Test2"],
            externalJobTitles: ["Test", "Test2"]
        },
        layoutState: {
            isNavigationExpanded: false
        },
        lookupsState: {
            providerList: [
                {
                    firstName: "string",
                    specialty: "string",
                    homeDepartment: "string",
                    specialtyId: 0,
                    schedulingName: "string",
                    providerTypeId: "string",
                    billable: false,
                    displayName: "string",
                    ansiNameCode: "string",
                    lastName: "string",
                    id: 0,
                    providerUserName: "string",
                    ansiSpecialtyCode: "string",
                    hideInPortal: true,
                    sex: "string",
                    entityType: 0,
                    providerType: "string",
                    createEncounterOnCheckin: false
                },{
                    firstName: "string",
                    specialty: "string",
                    homeDepartment: "string",
                    specialtyId: 1,
                    schedulingName: "string",
                    providerTypeId: "string",
                    billable: false,
                    displayName: "string",
                    ansiNameCode: "string",
                    lastName: "string",
                    id: 1,
                    providerUserName: "string",
                    ansiSpecialtyCode: "string",
                    hideInPortal: true,
                    sex: "string",
                    entityType: 0,
                    providerType: "string",
                    createEncounterOnCheckin: false
                }
            ],
            roleList: [
                {
                    name: "string0",
                    description: "string0",
                },{
                    name: "string1",
                    description: "string1",
                }
            ]
        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        },
        snackbarState: {
            messages: []
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders UserBulkActionStripe-Selection correctly checked", async () => {
        let fragment: any;
        
        await act(async () => {
            const {asFragment, getByTestId} = render(<TestWrapper mockState={mockState}>
                <UserBulkActionStripe currentStep={BulkAddStep.Selection}/>          
            </TestWrapper>);
            fragment = asFragment;
            await waitFor(() => {
                fireEvent.input(getByTestId('search'), {target: {value: 'checked'}});
            });
        });
        expect(fragment()).toMatchSnapshot();
    });

    it("renders UserBulkActionStripe-Selection correctly", async () => {
        let fragment: any;
        
        await act(async () => {
            const {asFragment, getByTestId} = render(<TestWrapper mockState={mockState}>
                <UserBulkActionStripe currentStep={BulkAddStep.Selection}/>          
            </TestWrapper>);
            fragment = asFragment;
            await waitFor(() => {
                fireEvent.click(getByTestId('bulk-check-all'));
            });
        });
        expect(fragment()).toMatchSnapshot();
    });

    it("renders UserBulkActionStripe-RolePicking correctly", async () => {
        let fragment: any;
        
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <UserBulkActionStripe currentStep={BulkAddStep.RolePicking}/>          
            </TestWrapper>);
            fragment = asFragment;
        });
        expect(fragment()).toMatchSnapshot();
    });

    it("renders UserBulkActionStripe-Review correctly", async () => {
        let fragment: any;
        
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <UserBulkActionStripe currentStep={BulkAddStep.Review}/>          
            </TestWrapper>);
            fragment = asFragment;
        });
        expect(fragment()).toMatchSnapshot();
    });

    it("renders UserBulkActionStripe-ProviderMapping correctly", async () => {
        let fragment: any;
        
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <UserBulkActionStripe currentStep={BulkAddStep.ProviderMapping}/>          
            </TestWrapper>);
            fragment = asFragment;
        });
        expect(fragment()).toMatchSnapshot();
    });
})