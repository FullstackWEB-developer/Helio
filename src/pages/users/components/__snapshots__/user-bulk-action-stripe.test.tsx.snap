// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBulkActionStripe tests renders UserBulkActionStripe-ProviderMapping correctly 1`] = `
<DocumentFragment>
  <div
    class="flex h-14 px-4 border-t border-b"
  >
    <div
      class="md:w-1/6 truncate"
    >
      <div
        class="subtitle2 pt-4"
      >
        users.bulk_section.x_users_selected_singular
      </div>
    </div>
    <div
      class="md:w-1/6 border-r border-l"
    >
      <div
        class="relative h-10 w-full search-input-field flex items-center relative h-full w-1/6"
      >
        <div
          class="absolute pl-4"
        >
          <div
            class=""
          >
            <svg
              class="cursor-pointer icon-small "
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="16"
                width="16"
              />
              <path
                class="search-icon-fill"
                d="M12.7,11.23a6.777,6.777,0,0,0,1.4-4.174A7.02,7.02,0,0,0,7.1,0,7.105,7.105,0,0,0,0,7.056a7.105,7.105,0,0,0,7.1,7.056,6.667,6.667,0,0,0,4.2-1.391l3,2.981a.971.971,0,0,0,1.4,0,.957.957,0,0,0,0-1.391Zm-5.6.8A5.022,5.022,0,0,1,2,7.056a5.1,5.1,0,0,1,10.2,0A5.022,5.022,0,0,1,7.1,12.025Z"
              />
            </svg>
          </div>
        </div>
        <input
          autocomplete="off"
          class="py-2.5 h-full w-full search-input-field body2 pl-12"
          data-testid="search"
          placeholder="users.search"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="flex items-center pl-6 body2"
    >
      <div
        class="relative"
      >
        <div
          class=""
        >
          <svg
            class="icon-medium cursor-pointer mr-4 "
            data-testid="svg-icon"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="24"
              width="24"
            />
            <path
              class="filter-icon"
              d="M10,18h4V16H10ZM3,6V8H21V6Zm3,7H18V11H6Z"
            />
          </svg>
        </div>
      </div>
      common.filters
    </div>
  </div>
</DocumentFragment>
`;

exports[`UserBulkActionStripe tests renders UserBulkActionStripe-Review correctly 1`] = `
<DocumentFragment>
  <div
    class="flex h-14 px-4 border-t border-b"
  >
    <div
      class="md:w-1/6 truncate"
    >
      <div
        class="subtitle2 pt-4"
      >
        users.bulk_section.x_users_selected_singular
      </div>
    </div>
    <div
      class="md:w-1/6 border-r border-l"
    >
      <div
        class="relative h-10 w-full search-input-field flex items-center relative h-full w-1/6"
      >
        <div
          class="absolute pl-4"
        >
          <div
            class=""
          >
            <svg
              class="cursor-pointer icon-small "
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="16"
                width="16"
              />
              <path
                class="search-icon-fill"
                d="M12.7,11.23a6.777,6.777,0,0,0,1.4-4.174A7.02,7.02,0,0,0,7.1,0,7.105,7.105,0,0,0,0,7.056a7.105,7.105,0,0,0,7.1,7.056,6.667,6.667,0,0,0,4.2-1.391l3,2.981a.971.971,0,0,0,1.4,0,.957.957,0,0,0,0-1.391Zm-5.6.8A5.022,5.022,0,0,1,2,7.056a5.1,5.1,0,0,1,10.2,0A5.022,5.022,0,0,1,7.1,12.025Z"
              />
            </svg>
          </div>
        </div>
        <input
          autocomplete="off"
          class="py-2.5 h-full w-full search-input-field body2 pl-12"
          data-testid="search"
          placeholder="users.search"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="flex items-center pl-6 body2"
    >
      <div
        class="relative"
      >
        <div
          class=""
        >
          <svg
            class="icon-medium cursor-pointer mr-4 "
            data-testid="svg-icon"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="24"
              width="24"
            />
            <path
              class="filter-icon"
              d="M10,18h4V16H10ZM3,6V8H21V6Zm3,7H18V11H6Z"
            />
          </svg>
        </div>
      </div>
      common.filters
    </div>
  </div>
</DocumentFragment>
`;

exports[`UserBulkActionStripe tests renders UserBulkActionStripe-RolePicking correctly 1`] = `
<DocumentFragment>
  <div
    class="flex h-14 px-4 border-t border-b"
  >
    <div
      class="md:w-1/6 truncate"
    >
      <div
        class="subtitle2 pt-4"
      >
        users.bulk_section.x_users_selected_singular
      </div>
    </div>
    <div
      class="md:w-1/6 border-r border-l"
    >
      <div
        class="relative h-10 w-full search-input-field flex items-center relative h-full w-1/6"
      >
        <div
          class="absolute pl-4"
        >
          <div
            class=""
          >
            <svg
              class="cursor-pointer icon-small "
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="16"
                width="16"
              />
              <path
                class="search-icon-fill"
                d="M12.7,11.23a6.777,6.777,0,0,0,1.4-4.174A7.02,7.02,0,0,0,7.1,0,7.105,7.105,0,0,0,0,7.056a7.105,7.105,0,0,0,7.1,7.056,6.667,6.667,0,0,0,4.2-1.391l3,2.981a.971.971,0,0,0,1.4,0,.957.957,0,0,0,0-1.391Zm-5.6.8A5.022,5.022,0,0,1,2,7.056a5.1,5.1,0,0,1,10.2,0A5.022,5.022,0,0,1,7.1,12.025Z"
              />
            </svg>
          </div>
        </div>
        <input
          autocomplete="off"
          class="py-2.5 h-full w-full search-input-field body2 pl-12"
          data-testid="search"
          placeholder="users.search"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="flex items-center pl-6 body2"
    >
      <div
        class="relative"
      >
        <div
          class=""
        >
          <svg
            class="icon-medium cursor-pointer mr-4 "
            data-testid="svg-icon"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="24"
              width="24"
            />
            <path
              class="filter-icon"
              d="M10,18h4V16H10ZM3,6V8H21V6Zm3,7H18V11H6Z"
            />
          </svg>
        </div>
      </div>
      common.filters
    </div>
  </div>
</DocumentFragment>
`;

exports[`UserBulkActionStripe tests renders UserBulkActionStripe-Selection correctly 1`] = `
<DocumentFragment>
  <div
    class="flex h-14 px-4 border-t border-b"
  >
    <div
      class="md:w-1/6 truncate"
    >
      <div
        class="h-9 pt-4"
      >
        <label
          class="flex flex-row items-center checkbox-button"
        >
          <input
            class="checkbox"
            data-testid="bulk-check-all"
            id="bulk-check-all_"
            name="bulk-check-all"
            type="checkbox"
          />
          <span
            class="checkbox-control"
          >
            <div
              class=""
            >
              <svg
                class="icon-medium "
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="24"
                  width="24"
                />
                <path
                  class="svg-checkbox"
                  d="M16,18H2a2,2,0,0,1-2-2V2A2,2,0,0,1,2,0H16a2,2,0,0,1,2,2V16A2,2,0,0,1,16,18ZM3.41,7.59h0L2,9l5,5,9-9L14.59,3.58,7,11.17,3.41,7.59Z"
                />
              </svg>
            </div>
          </span>
          <div
            class="truncate relative w-full h-full inline-block"
          >
            <span
              class="w-60 w-60"
            >
              users.bulk_section.x_users_selected_singular
            </span>
          </div>
        </label>
      </div>
    </div>
    <div
      class="md:w-1/6 border-r border-l"
    >
      <div
        class="relative h-10 w-full search-input-field flex items-center relative h-full w-1/6"
      >
        <div
          class="absolute pl-4"
        >
          <div
            class=""
          >
            <svg
              class="cursor-pointer icon-small "
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="16"
                width="16"
              />
              <path
                class="search-icon-fill"
                d="M12.7,11.23a6.777,6.777,0,0,0,1.4-4.174A7.02,7.02,0,0,0,7.1,0,7.105,7.105,0,0,0,0,7.056a7.105,7.105,0,0,0,7.1,7.056,6.667,6.667,0,0,0,4.2-1.391l3,2.981a.971.971,0,0,0,1.4,0,.957.957,0,0,0,0-1.391Zm-5.6.8A5.022,5.022,0,0,1,2,7.056a5.1,5.1,0,0,1,10.2,0A5.022,5.022,0,0,1,7.1,12.025Z"
              />
            </svg>
          </div>
        </div>
        <input
          autocomplete="off"
          class="py-2.5 h-full w-full search-input-field body2 pl-12"
          data-testid="search"
          placeholder="users.search"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="flex items-center pl-6 body2"
    >
      <div
        class="relative"
      >
        <div
          class=""
        >
          <svg
            class="icon-medium cursor-pointer mr-4 "
            data-testid="svg-icon"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="24"
              width="24"
            />
            <path
              class="filter-icon"
              d="M10,18h4V16H10ZM3,6V8H21V6Zm3,7H18V11H6Z"
            />
          </svg>
        </div>
      </div>
      common.filters
    </div>
  </div>
</DocumentFragment>
`;

exports[`UserBulkActionStripe tests renders UserBulkActionStripe-Selection correctly checked 1`] = `
<DocumentFragment>
  <div
    class="flex h-14 px-4 border-t border-b"
  >
    <div
      class="md:w-1/6 truncate"
    >
      <div
        class="h-9 pt-4"
      >
        <label
          class="flex flex-row items-center checkbox-button"
        >
          <input
            class="checkbox"
            data-testid="bulk-check-all"
            id="bulk-check-all_"
            name="bulk-check-all"
            type="checkbox"
          />
          <span
            class="checkbox-control"
          >
            <div
              class=""
            >
              <svg
                class="icon-medium "
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="24"
                  width="24"
                />
                <path
                  class="svg-checkbox"
                  d="M16,18H2a2,2,0,0,1-2-2V2A2,2,0,0,1,2,0H16a2,2,0,0,1,2,2V16A2,2,0,0,1,16,18ZM3.41,7.59h0L2,9l5,5,9-9L14.59,3.58,7,11.17,3.41,7.59Z"
                />
              </svg>
            </div>
          </span>
          <div
            class="truncate relative w-full h-full inline-block"
          >
            <span
              class="w-60 w-60"
            >
              users.bulk_section.x_users_selected_singular
            </span>
          </div>
        </label>
      </div>
    </div>
    <div
      class="md:w-1/6 border-r border-l"
    >
      <div
        class="relative h-10 w-full search-input-field flex items-center relative h-full w-1/6"
      >
        <div
          class="absolute pl-4"
        >
          <div
            class=""
          >
            <svg
              class="cursor-pointer icon-small "
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="16"
                width="16"
              />
              <path
                class="search-icon-fill"
                d="M12.7,11.23a6.777,6.777,0,0,0,1.4-4.174A7.02,7.02,0,0,0,7.1,0,7.105,7.105,0,0,0,0,7.056a7.105,7.105,0,0,0,7.1,7.056,6.667,6.667,0,0,0,4.2-1.391l3,2.981a.971.971,0,0,0,1.4,0,.957.957,0,0,0,0-1.391Zm-5.6.8A5.022,5.022,0,0,1,2,7.056a5.1,5.1,0,0,1,10.2,0A5.022,5.022,0,0,1,7.1,12.025Z"
              />
            </svg>
          </div>
        </div>
        <input
          autocomplete="off"
          class="py-2.5 h-full w-full search-input-field body2 pl-12"
          data-testid="search"
          placeholder="users.search"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="flex items-center pl-6 body2"
    >
      <div
        class="relative"
      >
        <div
          class=""
        >
          <svg
            class="icon-medium cursor-pointer mr-4 "
            data-testid="svg-icon"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="24"
              width="24"
            />
            <path
              class="filter-icon"
              d="M10,18h4V16H10ZM3,6V8H21V6Zm3,7H18V11H6Z"
            />
          </svg>
        </div>
      </div>
      common.filters
    </div>
  </div>
</DocumentFragment>
`;
